### PR TITLE
Fixes to xrootd logging config

### PR DESCRIPTION
**Pull request by joereuss12**
Changes include:
- Xrootd logging now uses the standard logrus logging values
- Instead of string comparisons for mapping, we now use log.ParseLevel and compare the levels (which are unsigned ints)
- The program now exits when there is an improper config value and returns a proper error
- Anywhere these configs are listed, they are all listed alphabetically for consistency and an easier time finding each option
- Unit tests for the generation of the xrootd config file

### CODE REVIEW

**Comment by joereuss12** [comment]
Let me know if I missed anything or need to add anything else! Hopefully this patches everything up.

File: xrootd/xrootd_config.go [hunk 1]
```diff
@@ -699,154 +704,308 @@ func SetUpMonitoring(ctx context.Context, egrp *errgroup.Group) error {
...
+	/////////////////////////ORIGIN/////////////////////////////
+	// Origin Cms
+	// https://xrootd.slac.stanford.edu/doc/dev54/cms_config.htm
+	originCmsConfig, err := log.ParseLevel(param.Logging_Origin_Cms.GetString())
+	if err != nil {
+		return errors.Wrap(err, "Error parsing specified log level for Origin_Cms, proper values include: panic, fatal, error, warn, info, debug, trace")
+	}
+	switch originCmsConfig {
+	case log.PanicLevel:
+		xrdConfig.Logging.OriginCms = "-all"
```
Comment (bbockelm) [inline xrootd/xrootd_config.go:718]: Use `fallthrough` for duplicate lines.

Doesn't save lines of code -- but does show intent (that things are supposed to be duplicated).

Comment (bbockelm) [reply xrootd/xrootd_config.go:718]: Actually, better yet, let's wrap this all into a function.  Here's an idea:

- Create a struct called `loggingMap` or similar.  Have each of the logging levels be a separate member of the struct.
- Iterate from top to bottom of the struct (in order of the logging levels).  If the string is set, use that in the xrdConfig.  If it's not set, use the prior value.
- Have a single helper function `genLoggingConfig` that does all the translation, including error messages.

`loggingMap` might look like:

```
type loggingMap struct {
    Trace string
    Debug string
    ...
    Panic string
}
```

So, in this case, we might see:

```
xrdConfig.Logging.OriginCms, err = genLoggingConfig(
    param.Logging_Origin_Cms.GetString(),
    loggingMap {
        Trace: "debug",
        Info: "all",
        Error: "-all",
    }
)
if err != nil {
    ...
}
```

That's about as compact as I can think of making this.

**Review by bbockelm — CHANGES_REQUESTED** [review]
Two things:
- See comments for one potential way to rewrite the `mapXrootdLogLevels` to be a bit more readable.
- Talk with @matyasselmeci about the description of the configuration parameters.  I feel they're still not particularly approachable.  How to write them to make sense to someone with little knowledge about XRootD?

**Comment by joereuss12** [comment]
Note: still need to make the config params more approachable and user-friendly, but thought I'd push my work in progress

**Review by bbockelm — CHANGES_REQUESTED** [review]
Very close.

I like the new, much more compact, style of setting the log levels.  It makes the code quite readable.

Only thing I spotted were some logging categories don't really make sense (e.g., setting logging for cache-related modules in the origin config).

File: xrootd/ca-bundle.crt
Comment (haoming29) [inline xrootd/ca-bundle.crt:1]: Are we intended to include these CAs in git? and why there's a CA update in log config PR?

**Review by haoming29 — CHANGES_REQUESTED** [review]
One small question and we should be good to go

File: xrootd/ca-bundle.crt
Comment (joereuss12) [reply xrootd/ca-bundle.crt:1]: Oh, not sure how I missed these getting added. I'm assuming they should not be there. I can try to remove them

**Comment by joereuss12** [comment]
Okay @haoming29 cleanup has happened. Just waiting for tests to pass to make sure I didn't remove anything important but I think everything should be good to go!

**Review by haoming29 — APPROVED** [review]
LGTM with the extra files removed. Thanks for taking care of this!

**Comment by haoming29** [comment]
@joereuss12 You might need to ask @jhiemstrawisc to overwrite Brian's requested change to get this merged in

**Comment by jhiemstrawisc** [comment]
@Haoming, since you've been reviewing for Brian, can you resolve any of the the conversations that are open that have a fix? Once I see everything resolved, I'll merge

**Comment by haoming29** [comment]
> @haoming, since you've been reviewing for Brian, can you resolve any of the the conversations that are open that have a fix? Once I see everything resolved, I'll merge

@jhiemstrawisc  Peek through the conversations and looks to me they have all been taken care of. Resolved conversations accordingly

**Comment by jhiemstrawisc** [comment]
Perfect, thanks. Just wanted the acknowledgement from someone for bookkeeping since I'm not reviewing myself :) Consider it merged! Thanks to everyone who worked on this

**Pull request merged**